bin/list-all

```diff
@@ -8,9 +8,7 @@ source "$(dirname $0)/../lib/utils.sh"
 list_versions() {
   ensure_ruby_build_setup
 
-  # This sed command was a quick and dirty solution to remove topaz-dev from
-  # the version list, since it doesn't exist
-  "$(ruby_build_path)" --definitions | paste -sd " " - | sed "s|topaz-dev||g"
+  "$(ruby_build_path)" --definitions | grep -v "topaz-dev" | paste -sd " " -
 }
 
-list_versions
+echo $(list_versions)
```

Comment:
Did shellcheck complain about this line? I don't think the echo is needed here. Just `list_versions` should be enough.

Reply:
Actually, it complains:
```
$ shellcheck bin/list-all

In bin/list-all line 6:
source "$(dirname $0)/../lib/utils.sh"
       ^-----------------------------^ SC1091: Not following: ../lib/utils.sh was not specified as input (see shellcheck -x).
                  ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
source "$(dirname "$0")/../lib/utils.sh"


In bin/list-all line 14:
echo $(list_versions)
     ^--------------^ SC2046: Quote this to prevent word splitting.
     ^--------------^ SC2005: Useless echo? Instead of 'echo $(cmd)', just use 'cmd'.

For more information:
  https://www.shellcheck.net/wiki/SC2046 -- Quote this to prevent word splitt...
  https://www.shellcheck.net/wiki/SC1091 -- Not following: ../lib/utils.sh wa...
  https://www.shellcheck.net/wiki/SC2086 -- Double quote to prevent globbing ...
```

But without it nothing is printed:

```
~/.asdf/plugins/ruby (master $ u=)
$ asdf list-all ruby
Downloading ruby-build...
Cloning into 'ruby-build-source'...
remote: Enumerating objects: 129, done.
remote: Counting objects: 100% (129/129), done.
remote: Compressing objects: 100% (89/89), done.
remote: Total 10073 (delta 73), reused 80 (delta 36), pack-reused 9944
Receiving objects: 100% (10073/10073), 2.15 MiB | 4.06 MiB/s, done.
Resolving deltas: 100% (6525/6525), done.
HEAD is now at f6e8dc1 ruby-build 20191102
/Users/kassio/.asdf/plugins/ruby/ruby-build-source
~/.asdf/plugins/ruby (master $ u=)
```

My shell:
```
$ zsh --version
zsh --version
zsh 5.7.1 (x86_64-apple-darwin19.0.0)
```

Reply:
I think you need to direct STDERR to `/dev/null` to solve this. Try this instead:

```
list_versions 2> /dev/null
```

Reply:
Unfortunately it didn't work.

![_HOME__asdf_plugins_ruby__zsh_](https://user-images.githubusercontent.com/120483/69064420-140a5080-0a16-11ea-983c-37684f0fe6e2.png)

Reply:
This works on my machine:

```
#!/usr/bin/env bash

set -euo pipefail

# shellcheck source=../lib/utils.sh
source "$(dirname $0)/../lib/utils.sh"

list_versions() {
  ensure_ruby_build_setup

  # This sed command was a quick and dirty solution to remove topaz-dev from
  # the version list, since it doesn't exist
  "$(ruby_build_path)" --definitions | grep -v "topaz-dev" | paste -sd " " -
}

list_versions 2> /dev/null
```

It looks like you reverted your other change on line 11 when you tested that.

Reply:
Just to be sure, are you on zsh?

Reply:
Yes I am, but those changes work for me in Zsh and Bash.

Reply:
Ok, I'll investigate why my shell is not working.

Changed the code to be like you suggested.

Reply:
So, it seems that my problem is related to `$CDPATH` which I suppose maybe a problem with `asdf` itself.

```console
~/.asdf/plugins/ruby (master u+1)
$ echo $CDPATH
.:/Users/kassio:/Users/kassio/src:/Users/kassio/.config:

~/.asdf/plugins/ruby (master u+1)
$ asdf list-all ruby
/Users/kassio/.asdf/plugins/ruby/ruby-build-source

~/.asdf/plugins/ruby (master u+1)
$ CDPATH="" asdf list-all ruby
1.8.5-p52
1.8.5-p113
1.8.5-p114
1.8.5-p115
1.8.5-p231
1.8.6
# ...
```

Reply:
On the other hand, [asdf elixir plugin works and it uses `echo`](https://github.com/asdf-vm/asdf-elixir/blob/master/bin/list-all).

Reply:
Are you exporting `CDPATH` or just setting it? Because it can meddle with how the path is interpreted, and also cause `cd` to print to STDOUT, you should not export `CDPATH`. Here is what I have in my dotfiles - https://github.com/Stratus3D/dotfiles/blob/44678c7c645ae9a442b4fa246cfd39b9e4d0a74c/mixins/general#L28-L31

Reply:
I tried without exporting the CDPATH and still doesn't work on my environment.
What do you think of my proposal to use the full path for the `ruby-build-source`?

Reply:
Full path should be fine.